tests/infer/test_hmc.py

```diff
@@ -0,0 +1,116 @@
+from __future__ import absolute_import, division, print_function
+
+import logging
+from collections import defaultdict
+
+import pytest
+import torch
+from torch.autograd import Variable
+
+import pyro
+import pyro.distributions as dist
+from pyro.infer.mcmc.hmc import HMC
+from pyro.infer.mcmc.mcmc import MCMC
+from pyro.infer.mcmc.verlet_integrator import verlet_integrator
+from tests.common import assert_equal
+
+logging.basicConfig(format='%(levelname)s %(message)s')
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class GaussianChain(object):
```

Comment:
These are simply borrowed from @martinjankowiak's conjugate gaussian tests, albeit not testing in all its generality. It will be nice to refactor them so that they are can be reused across different algorithms for comparison.